src/Microsoft.Data.SqlClient/src/Microsoft.Data.SqlClient.csproj

```diff
@@ -1,55 +1,108 @@
+    <!-- included into the build.                                                                 -->
     <EmbeddedResource Include="Resources/ILLink.Substitutions.xml"
-                      Condition="'$(TargetOs.ToUpper())' == 'WINDOWS_NT'" />
+                      Condition="'$(NormalizedTargetOs)' == 'windows_nt' AND '$(TargetFramework)' != 'net462'" />
   </ItemGroup>
 
   <!-- References ====================================================== -->
-  <ItemGroup>
-    <!-- References that apply to all target frameworks -->
+  <!-- References for netframework -->
+  <ItemGroup Condition="'$(TargetFramework)' == 'net462'">
+    <Reference Include="System.Configuration" />
+    <Reference Include="System.EnterpriseServices" />
+    <Reference Include="System.Transactions" />
+    
     <PackageReference Include="Azure.Core" />
     <PackageReference Include="Azure.Identity" />
     <PackageReference Include="Microsoft.Bcl.Cryptography" />
+    <PackageReference Include="Microsoft.Data.SqlClient.SNI">
+      <PrivateAssets>All</PrivateAssets>
+      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+    </PackageReference>
     <PackageReference Include="Microsoft.Extensions.Caching.Memory" />
     <PackageReference Include="Microsoft.IdentityModel.JsonWebTokens" />
     <PackageReference Include="Microsoft.IdentityModel.Protocols.OpenIdConnect" />
-    <PackageReference Include="System.Configuration.ConfigurationManager" />
-    <PackageReference Include="System.Security.Cryptography.Pkcs"  />
-  </ItemGroup>
-  <ItemGroup Condition="'$(TargetFramework)' == 'net462'">
-    <!-- References that only apply to net462 -->
-    <Reference Include="System.Configuration" />
-    <Reference Include="System.Transactions" />
-    <PackageReference Include="System.ValueTuple" />
-    <PackageReference Include="System.Threading.Channels" />
-	</ItemGroup>
-
-  <!-- System.Buffers, System.Memory, and System.Runtime.InteropServices.RuntimeInformation are included in .NET 10+ BCL -->
-  <ItemGroup Condition="$([MSBuild]::VersionLessThan($([MSBuild]::GetTargetFrameworkVersion($(TargetFramework))), '10.0'))">
     <PackageReference Include="System.Buffers" />
+    <PackageReference Include="System.Diagnostics.DiagnosticSource" />
     <PackageReference Include="System.Memory" />
     <PackageReference Include="System.Runtime.InteropServices.RuntimeInformation" />
+    <PackageReference Include="System.Security.Cryptography.Pkcs" />
+    <PackageReference Include="System.Text.Json" />
+    <PackageReference Include="System.Threading.Channels" />
+    <PackageReference Include="System.ValueTuple" />
+	</ItemGroup>
+
+  <!-- References for netcore -->
+  <ItemGroup Condition="'$(TargetFramework)' != 'net462'">
+    <PackageReference Include="Azure.Core" />
+    <PackageReference Include="Azure.Identity" />
+    <PackageReference Include="Microsoft.Bcl.Cryptography" />
```

Comment:
Please also include dependency on Microsoft.Identity.client (MSAL).

It should be present in all projects, I don't recall why it got removed. But, we use its APIs directly in AAD implementation and cannot connect with AAD Auth modes when debugging locally unless its included.